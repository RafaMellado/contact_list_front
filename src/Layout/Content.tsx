import React from "react";

import "./Styles/Content.css";

export interface ContentProps {
  children: React.ReactChild;
}

export function Content({ children }: ContentProps) {
  return <div className="content">{children}</div>;
}
