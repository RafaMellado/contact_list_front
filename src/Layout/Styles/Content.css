.content {
  margin-top: 2rem;
  min-height: calc(100vh - var(--header-height));
}

@media screen and (min-width: 1024px) {
  .content {
    min-height: calc(100vh - var(--header-height));
    padding: 4rem;
  }
}
