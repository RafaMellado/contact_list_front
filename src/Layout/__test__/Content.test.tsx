import { Content } from "../Content";
import { act, render } from "@testing-library/react";
import type { ContentProps } from "../Content";

describe(`<${Content.name} />`, () => {
  const contentProps: ContentProps = {
    children: <div>test</div>,
  };

  const factoryComponent = (props: ContentProps = contentProps) =>
    render(<Content {...props} />);

  test("renders correctly the component", () => {
    const component = factoryComponent();

    act(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
