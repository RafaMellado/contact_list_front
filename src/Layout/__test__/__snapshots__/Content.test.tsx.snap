// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> renders correctly the component 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    <div>
      test
    </div>
  </div>
</DocumentFragment>
`;
